### ISSUE TITLE
Fedora 35 compatibility

### ISSUE THREAD

**Issue by e7d** [timothycrosley/streamdeck-ui#166]
Hello,

After updating to Fedora 35 this morning, my streamdeck_ui was broken.
Reinstalling it gave me a bunch of successive errors because of missing dependencies on my system.

I managed to get it to run again by making sure all of the following dev packages were installed:
- freetype-devel
- hidapi-devel
- libhid-devel
- libjpeg-devel
- libpng-devel
- libusb-devel
- libxc-devel
- systemd-devel
- zlib-devel

Not sure all of them are required, but here you go. 

As far as I can understand what happened, some of these dependencies would be directly needed by streamedck_ui (hidapi, libhid, libusb, libxc, zlib) while the others are needed by a Python peer dependency named Pillow (freetype-devel, libjpeg-devel, libpng-devel).

To make all of that work, I had to make some cleanup on my system, especially since Pillow was not in a good place.
Please find below a script that makes streameck_ui work 100% on my Fedora 35, taking care of all the needed steps:
```sh
pip3 uninstall pil Pillow streamdeck_ui
pip3 cache purge
sudo dnf install freetype-devel hidapi-devel libhid-devel libjpeg-devel libpng-devel libusb-devel libxc-devel systemd-devel zlib-devel
pip3 install --user streamdeck_ui
```

Hope it could help.
Cheers!

**Comment by genebean** [comment]
The suggested commands above made it so I was able to install too.

**Comment by patrickhgh** [comment]
I followed the steps above and I am still having issues.

streamdeck
Traceback (most recent call last):
  File "/home/user/.local/bin/streamdeck", line 8, in <module>
    sys.exit(start())
  File "/home/user/.local/lib/python3.10/site-packages/streamdeck_ui/gui.py", line 269, in start
    build_device(ui)
  File "/home/user/.local/lib/python3.10/site-packages/streamdeck_ui/gui.py", line 199, in build_device
    build_buttons(ui, page)
  File "/home/user/.local/lib/python3.10/site-packages/streamdeck_ui/gui.py", line 129, in build_buttons
    deck = api.get_deck(deck_id)
  File "/home/user/.local/lib/python3.10/site-packages/streamdeck_ui/api.py", line 133, in get_deck
    return {"type": decks[deck_id].deck_type(), "layout": decks[deck_id].key_layout()}
KeyError: None

**Comment by TekNinjaUK** [comment]
I am also seeing the exact errors line for line as above.

Reinstalled all dependencies, rebooted, unplug/replug the SD. Folled the steps above also.

Fully patched Fedora 35.

```
uname -a
Linux HOSTNAME 5.15.12-200.fc35.x86_64 #1 SMP Wed Dec 29 15:03:38 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by daltonone** [comment]
I also followed the above steps and am having the same error.

**Comment by daltonone** [comment]
I found out some more information about why mine wasn’t working. I purchased the Mark 2 not V2 which has a different pid.

**Comment by daltonone** [comment]
This is a python3.10 issue.  Large parts of the poetry install require python<3.10 and Fedora 35 comes with python3.10.1.  To solve this I forked the repo, cloned it to my local, installed python 3.9.6, setup a virtualenv running python3.9.6, and did the poetry install.  After that you can just run streamdeck.  This is all probably overkill, but I was downloading thinking I was looking for a way to add a new device and was pleasantly surprised.  Hope this helps someone else.

After forking:
```
git clone https://github.com/$GITHUB_ACCOUNT/streamdeck-ui.git
wget https://www.python.org/ftp/python/3.9.6/Python-3.9.6.tgz
tar xzf Python-3.9.6.tgz 
rm Python-3.9.6.tgz
cd Python-3.9.6 
sudo ./configure --enable-optimizations 
sudo make altinstall
python3.9 -V 
cd ../streamdeck-ui
virtualenv -p `which python3.9` ./streamdeck-env
source streamdeck-env/bin/activate
poetry install
```

**Comment by dodgyrabbit** [comment]
Version 1.1.3 is now published and should resolve the python 3.9 dependency issue as well as provide support for Mark 2 type devices.
